// vitepress-theme-teek@1.4.6/es/components/theme/Layout/src/index.vue2.mjs
import { defineComponent, provide, computed, ref, watch, createElementBlock, createBlock, unref, openBlock, Fragment, createCommentVNode, renderSlot, withDirectives, withCtx, normalizeProps, guardReactiveProps, createVNode, vShow, createSlots, renderList, normalizeClass, resolveDynamicComponent, mergeProps } from 'vue';
import DefaultTheme from 'vitepress/theme';
import { useData, onContentUpdated } from 'vitepress';
import { isBoolean, isClient } from '../../../../helper/dist/index.mjs';
import { localeContextKey } from '../../../../composables/useLocale.mjs';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import '../../../../composables/useZIndex.mjs';
import { useTeekConfig, usePageState } from '../../ConfigProvider/index.mjs';
import _sfc_main$v from '../../Home/src/index.vue2.mjs';
import _sfc_main$q from '../../HomeMyCard/src/HomeMyCardScreen.vue2.mjs';
import _sfc_main$8 from '../../BodyBgImage/src/index.vue2.mjs';
import _sfc_main$r from '../../FooterGroup/src/index.vue2.mjs';
import _sfc_main$s from '../../FooterInfo/src/index.vue2.mjs';
import _sfc_main$m from '../../ArticleImagePreview/src/index.vue2.mjs';
import _sfc_main$l from '../../ArticleAnalyze/src/index.vue2.mjs';
import _sfc_main$c from '../../ArticleShare/src/index.vue2.mjs';
import _sfc_main$h from '../../ArticleUpdate/src/index.vue2.mjs';
import _sfc_main$9 from '../../ArticleHeadingHighlight/src/index.vue2.mjs';
import _sfc_main$n from '../../ArticlePageStyle/src/index.vue2.mjs';
import _sfc_main$g from '../../ArticleAppreciation/src/AsideBottomAppreciation.vue2.mjs';
import _sfc_main$i from '../../ArticleAppreciation/src/DocAfterAppreciation.vue2.mjs';
import _sfc_main$j from '../../ArticleAppreciation/src/DocAfterAppreciationPopper.vue2.mjs';
import _sfc_main$4 from '../../CommentTwikoo/src/index.vue2.mjs';
import _sfc_main$1 from '../../CommentArtalk/src/index.vue2.mjs';
import _sfc_main$2 from '../../CommentGiscus/src/index.vue2.mjs';
import _sfc_main$3 from '../../CommentWaline/src/index.vue2.mjs';
import _sfc_main$o from '../../CodeBlockToggle/src/index.vue2.mjs';
import _sfc_main$b from '../../RightBottomButton/src/index.vue2.mjs';
import _sfc_main$a from '../../Notice/src/index.vue2.mjs';
import _sfc_main$t from '../../ThemeEnhance/src/index.vue2.mjs';
import _sfc_main$k from '../../../common/VpContainer/src/index.vue2.mjs';
import _sfc_main$d from '../../ArchivesPage/src/index.vue2.mjs';
import _sfc_main$e from '../../CataloguePage/src/index.vue2.mjs';
import _sfc_main$f from '../../ArticleOverviewPage/src/index.vue2.mjs';
import _sfc_main$6 from '../../LoginPage/src/index.vue2.mjs';
import { useWatchLogin } from '../../LoginPage/src/useWatchLogin.mjs';
import _sfc_main$7 from '../../RiskLinkPage/src/index.vue2.mjs';
import { useRiskLink } from '../../RiskLinkPage/src/useRiskLink.mjs';
import _sfc_main$p from '../../SidebarTrigger/src/index.vue2.mjs';
import _sfc_main$u from '../../HomeFeature/src/index.vue2.mjs';
import _sfc_main$5 from '../../RouteLoading/src/index.vue.mjs';

var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "TeekLayout" },
  __name: "index",
  props: {
    locale: {}
  },
  setup(__props) {
    const props = __props;
    provide(
      localeContextKey,
      computed(() => props.locale)
    );
    const { Layout } = DefaultTheme;
    const ns = useNamespace("layout");
    const { getTeekConfigRef } = useTeekConfig();
    const { isHomePage, isArchivesPage, isCataloguePage, isArticleOverviewPage } = usePageState();
    const { frontmatter, localeIndex, page } = useData();
    const teekConfig = getTeekConfigRef(null, {
      teekTheme: true,
      teekHome: true,
      vpHome: true,
      sidebarTrigger: false,
      loading: false,
      codeBlock: { enabled: true },
      themeSize: "",
      bodyBgImg: {},
      notice: {},
      comment: { provider: "" },
      articleUpdate: { enabled: true },
      articleTopTip: void 0,
      articleBottomTip: void 0,
      articleShare: {},
      appreciation: {},
      riskLink: { enabled: false },
      themeEnhance: { enabled: true }
    });
    const loading = ref(teekConfig.value.loading);
    const commentConfig = computed(() => {
      const comment = frontmatter.value.comment ?? teekConfig.value.comment;
      if (isBoolean(comment)) return { enabled: comment };
      return {
        enabled: true,
        components: {
          twikoo: _sfc_main$4,
          waline: _sfc_main$3,
          giscus: _sfc_main$2,
          artalk: _sfc_main$1
        },
        provider: comment.provider,
        options: comment.options
      };
    });
    const topTipConfig = computed(() => {
      if (isBoolean(teekConfig.value.articleTopTip)) return teekConfig.value.articleTopTip;
      return teekConfig.value.articleTopTip?.(frontmatter.value, localeIndex.value, page.value);
    });
    const bottomTipConfig = computed(() => {
      if (isBoolean(teekConfig.value.articleBottomTip)) return teekConfig.value.articleBottomTip;
      return teekConfig.value.articleBottomTip?.(frontmatter.value, localeIndex.value, page.value);
    });
    const themeSizeAttribute = ns.join("theme-size");
    watch(
      () => teekConfig.value.themeSize,
      (newValue) => {
        if (!isClient) return;
        if (newValue) document.documentElement.setAttribute(themeSizeAttribute, newValue);
        else document.documentElement.removeAttribute(themeSizeAttribute);
      },
      { immediate: true, flush: "post" }
    );
    const { watchSite, watchPages } = useWatchLogin();
    const { restart } = useRiskLink({
      whitelist: teekConfig.value.riskLink.whitelist,
      blacklist: teekConfig.value.riskLink.blacklist
    });
    watchSite();
    watchPages();
    onContentUpdated(() => {
      if (teekConfig.value.riskLink.enabled) restart();
    });
    const usedSlots = [
      "home-hero-before",
      "home-features-after",
      "nav-bar-content-after",
      "layout-bottom",
      "doc-footer-before",
      "doc-before",
      "doc-after",
      "aside-bottom",
      "page-top",
      "aside-outline-before",
      "sidebar-nav-before"
    ];
    return (_ctx, _cache) => {
      return unref(teekConfig).teekTheme ? (openBlock(), createElementBlock(
        Fragment,
        { key: 0 },
        [
          unref(teekConfig).loading ?? false ? (openBlock(), createBlock(unref(_sfc_main$5), {
            key: 0,
            modelValue: loading.value,
            "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => loading.value = $event)
          }, {
            default: withCtx((scope) => [
              renderSlot(_ctx.$slots, "teek-loading", normalizeProps(guardReactiveProps(scope)))
            ]),
            _: 3
            /* FORWARDED */
          }, 8, ["modelValue"])) : createCommentVNode("v-if", true),
          unref(frontmatter).loginPage === true ? renderSlot(_ctx.$slots, "teek-login-page", { key: 1 }, () => [
            withDirectives(createVNode(
              unref(_sfc_main$6),
              null,
              null,
              512
              /* NEED_PATCH */
            ), [
              [vShow, !loading.value]
            ])
          ]) : createCommentVNode("v-if", true),
          unref(frontmatter).riskLinkPage === true ? renderSlot(_ctx.$slots, "teek-risk-link-page", { key: 2 }, () => [
            withDirectives(createVNode(
              unref(_sfc_main$7),
              null,
              null,
              512
              /* NEED_PATCH */
            ), [
              [vShow, !loading.value]
            ])
          ]) : createCommentVNode("v-if", true),
          unref(frontmatter).layout !== false ? (openBlock(), createElementBlock(
            Fragment,
            { key: 3 },
            [
              unref(teekConfig).bodyBgImg?.imgSrc ? (openBlock(), createBlock(unref(_sfc_main$8), { key: 0 })) : createCommentVNode("v-if", true),
              createVNode(unref(_sfc_main$9)),
              unref(teekConfig).notice?.enabled ? (openBlock(), createBlock(
                unref(_sfc_main$a),
                { key: 1 },
                createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: withCtx((scope) => [
                        renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )) : createCommentVNode("v-if", true),
              createVNode(
                unref(_sfc_main$b),
                null,
                createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: withCtx((scope) => [
                        renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )
            ],
            64
            /* STABLE_FRAGMENT */
          )) : createCommentVNode("v-if", true),
          withDirectives(createVNode(unref(Layout), {
            class: normalizeClass([
              unref(ns).b(),
              { [unref(ns).m("hide-vp-home")]: !unref(teekConfig).vpHome },
              unref(ns).has("sidebar-trigger", unref(teekConfig).sidebarTrigger)
            ])
          }, createSlots({
            "home-hero-before": withCtx(() => [
              renderSlot(_ctx.$slots, "home-hero-before"),
              renderSlot(_ctx.$slots, "teek-home-before"),
              createCommentVNode(" \u81EA\u5B9A\u4E49\u9996\u9875 "),
              unref(teekConfig).teekHome ? (openBlock(), createBlock(
                unref(_sfc_main$v),
                { key: 0 },
                createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: withCtx((scope) => [
                        renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-home-after")
            ]),
            "home-features-after": withCtx(() => [
              !unref(teekConfig).teekHome ? (openBlock(), createElementBlock(
                Fragment,
                { key: 0 },
                [
                  renderSlot(_ctx.$slots, "teek-home-features-before"),
                  createVNode(unref(_sfc_main$u)),
                  renderSlot(_ctx.$slots, "teek-home-features-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "home-features-after")
            ]),
            "nav-bar-content-after": withCtx(() => [
              renderSlot(_ctx.$slots, "nav-bar-content-after"),
              unref(teekConfig).themeEnhance.enabled ?? true ? (openBlock(), createBlock(
                unref(_sfc_main$t),
                { key: 0 },
                createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: withCtx((scope) => [
                        renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )) : createCommentVNode("v-if", true)
            ]),
            "layout-bottom": withCtx(() => [
              unref(isHomePage) ? (openBlock(), createBlock(unref(_sfc_main$r), { key: 0 })) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-footer-info-before"),
              renderSlot(_ctx.$slots, "teek-footer-info", {}, () => [
                unref(isHomePage) ? (openBlock(), createBlock(unref(_sfc_main$s), { key: 0 })) : createCommentVNode("v-if", true)
              ]),
              renderSlot(_ctx.$slots, "teek-footer-info-after"),
              renderSlot(_ctx.$slots, "layout-bottom")
            ]),
            "sidebar-nav-before": withCtx(() => [
              createVNode(unref(_sfc_main$q))
            ]),
            "doc-before": withCtx(() => [
              renderSlot(_ctx.$slots, "doc-before"),
              renderSlot(_ctx.$slots, "teek-article-analyze-before"),
              unref(frontmatter).article !== false ? (openBlock(), createBlock(unref(_sfc_main$l), { key: 0 })) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-article-analyze-after"),
              createVNode(unref(_sfc_main$m)),
              createVNode(unref(_sfc_main$n)),
              unref(teekConfig).codeBlock.enabled ?? true ? (openBlock(), createBlock(unref(_sfc_main$o), { key: 1 })) : createCommentVNode("v-if", true),
              topTipConfig.value ? (openBlock(), createBlock(
                unref(_sfc_main$k),
                normalizeProps(mergeProps({ key: 2 }, unref(isBoolean)(topTipConfig.value) ? {} : topTipConfig.value)),
                null,
                16
                /* FULL_PROPS */
              )) : createCommentVNode("v-if", true),
              unref(teekConfig).sidebarTrigger ? (openBlock(), createBlock(unref(_sfc_main$p), { key: 3 }, {
                default: withCtx((scope) => [
                  renderSlot(_ctx.$slots, "teek-sidebar-trigger", normalizeProps(guardReactiveProps(scope)))
                ]),
                _: 3
                /* FORWARDED */
              })) : createCommentVNode("v-if", true)
            ]),
            "doc-footer-before": withCtx(() => [
              renderSlot(_ctx.$slots, "doc-footer-before"),
              renderSlot(_ctx.$slots, "teek-article-bottom-tip-before"),
              bottomTipConfig.value ? (openBlock(), createBlock(
                unref(_sfc_main$k),
                normalizeProps(mergeProps({ key: 0 }, unref(isBoolean)(bottomTipConfig.value) ? {} : bottomTipConfig.value)),
                null,
                16
                /* FULL_PROPS */
              )) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-article-bottom-tip-after")
            ]),
            "doc-after": withCtx(() => [
              renderSlot(_ctx.$slots, "doc-after"),
              renderSlot(_ctx.$slots, "teek-doc-update-before"),
              (unref(teekConfig).articleUpdate.enabled ?? true) && unref(frontmatter).articleUpdate !== false ? (openBlock(), createBlock(unref(_sfc_main$h), { key: 0 })) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-doc-update-after"),
              renderSlot(_ctx.$slots, "teek-doc-after-appreciation-before"),
              unref(teekConfig).appreciation.position === "doc-after" ? (openBlock(), createBlock(unref(_sfc_main$i), { key: 1 })) : unref(teekConfig).appreciation.position === "doc-after-popper" ? (openBlock(), createBlock(unref(_sfc_main$j), { key: 2 })) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-doc-after-appreciation-after"),
              renderSlot(_ctx.$slots, "teek-comment-before"),
              createCommentVNode(" \u8BC4\u8BBA\u533A "),
              commentConfig.value.enabled && commentConfig.value.provider ? (openBlock(), createElementBlock(
                Fragment,
                { key: 3 },
                [
                  commentConfig.value.provider === "render" ? renderSlot(_ctx.$slots, "teek-comment", { key: 0 }) : (openBlock(), createBlock(resolveDynamicComponent(commentConfig.value.components?.[commentConfig.value.provider]), {
                    key: 1,
                    id: `${unref(ns).namespace}-comment`,
                    class: normalizeClass(unref(ns).e("comment"))
                  }, null, 8, ["id", "class"]))
                ],
                64
                /* STABLE_FRAGMENT */
              )) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-comment-after")
            ]),
            "aside-bottom": withCtx(() => [
              renderSlot(_ctx.$slots, "aside-bottom"),
              renderSlot(_ctx.$slots, "teek-aside-bottom-appreciation-before"),
              unref(teekConfig).appreciation.position === "aside-bottom" ? (openBlock(), createBlock(unref(_sfc_main$g), { key: 0 })) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-aside-bottom-appreciation-after")
            ]),
            "page-top": withCtx(() => [
              renderSlot(_ctx.$slots, "page-top"),
              renderSlot(_ctx.$slots, "teek-page-top-before"),
              unref(isArchivesPage) ? (openBlock(), createBlock(
                unref(_sfc_main$d),
                { key: 0 },
                createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: withCtx((scope) => [
                        renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )) : createCommentVNode("v-if", true),
              unref(isCataloguePage) ? (openBlock(), createBlock(
                unref(_sfc_main$e),
                { key: 1 },
                createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: withCtx((scope) => [
                        renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )) : createCommentVNode("v-if", true),
              unref(isArticleOverviewPage) ? (openBlock(), createBlock(unref(_sfc_main$f), { key: 2 })) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-page-top-after")
            ]),
            "aside-outline-before": withCtx(() => [
              renderSlot(_ctx.$slots, "teek-article-share-before"),
              unref(teekConfig).articleShare.enabled ? (openBlock(), createBlock(unref(_sfc_main$c), { key: 0 })) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-article-share-after"),
              renderSlot(_ctx.$slots, "aside-outline-before")
            ]),
            _: 2
            /* DYNAMIC */
          }, [
            renderList(Object.keys(_ctx.$slots).filter((name) => !usedSlots.includes(name)), (name) => {
              return {
                name,
                fn: withCtx((slotData) => [
                  renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(slotData)))
                ])
              };
            })
          ]), 1032, ["class"]), [
            [vShow, !loading.value]
          ])
        ],
        64
        /* STABLE_FRAGMENT */
      )) : (openBlock(), createBlock(
        unref(Layout),
        { key: 1 },
        createSlots({
          _: 2
          /* DYNAMIC */
        }, [
          renderList(_ctx.$slots, (_, name) => {
            return {
              name,
              fn: withCtx((slotData) => [
                renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(slotData)))
              ])
            };
          })
        ]),
        1024
        /* DYNAMIC_SLOTS */
      ));
    };
  }
});

export { _sfc_main as default };
