// vitepress-theme-teek@1.4.6/lib/components/common/Segmented/src/SegmentedItem.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('../../../../composables/useZIndex.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../Icon/src/index.vue2.js');
require('@iconify/vue');

const _hoisted_1 = ["title", "disabled"];
const _hoisted_2 = ["value", "name", "disabled", "checked", "aria-checked"];
const _hoisted_3 = { key: 1 };
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "SegmentedItem" },
  __name: "SegmentedItem",
  props: /* @__PURE__ */ Vue.mergeModels({
    disabled: { type: Boolean },
    value: { type: [String, Number, Object, Boolean] },
    label: {},
    icon: {},
    title: {},
    name: {}
  }, {
    "modelValue": { type: [String, Number, Object, Boolean] },
    "modelModifiers": {}
  }),
  emits: ["update:modelValue"],
  setup(__props) {
    const ns = useNamespace.useNamespace("segmented-item");
    const model = Vue.useModel(__props, "modelValue");
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("label", {
        class: Vue.normalizeClass([Vue.unref(ns).b(), Vue.unref(ns).is("active", model.value === _ctx.value && !_ctx.disabled), Vue.unref(ns).is("disabled", _ctx.disabled)]),
        title: _ctx.title,
        disabled: _ctx.disabled
      }, [
        Vue.withDirectives(Vue.createElementVNode("input", {
          "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => model.value = $event),
          type: "radio",
          value: _ctx.value,
          name: _ctx.name,
          disabled: _ctx.disabled,
          checked: model.value === _ctx.value,
          "aria-checked": model.value === _ctx.value,
          role: "radio",
          style: { "display": "none" }
        }, null, 8, _hoisted_2), [
          [Vue.vModelRadio, model.value]
        ]),
        Vue.createElementVNode(
          "span",
          {
            class: Vue.normalizeClass(Vue.unref(ns).e("content"))
          },
          [
            _ctx.icon ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
              key: 0,
              icon: _ctx.icon,
              "aria-hidden": "true"
            }, null, 8, ["icon"])) : Vue.createCommentVNode("v-if", true),
            _ctx.label ? (Vue.openBlock(), Vue.createElementBlock(
              "span",
              _hoisted_3,
              Vue.toDisplayString(_ctx.label),
              1
              /* TEXT */
            )) : Vue.createCommentVNode("v-if", true)
          ],
          2
          /* CLASS */
        )
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
